packages/gossiplog/src/AbstractGossipLog.ts

```diff
@@ -241,25 +243,26 @@ export abstract class AbstractGossipLog<Payload = unknown> extends TypedEventEmi
 			const signedMessage = this.encode(signature, message)
 			this.log("appending message %s at clock %d with parents %o", signedMessage.id, clock, parents)
 
-			const result = await this.apply(txn, signedMessage)
+			const applyResult = await this.apply(txn, signedMessage)
 
-			root = result.root
-			heads = result.heads
+			root = applyResult.root
+			heads = applyResult.heads
+			result = applyResult.result
 
 			return signedMessage
 		})
 
 		assert(root !== null && heads !== null, "failed to commit transaction")
 		this.dispatchEvent(new CustomEvent("commit", { detail: { root, heads } }))
 
-		return signedMessage
+		return { ...signedMessage, result }
```

Comment:
signedMessage is a `SignedMessage` class, you can't use the object spread operator with it